chain/bond.go

```diff
@@ -0,0 +1,134 @@
+// Copyright (C) 2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package chain
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/ava-labs/avalanchego/database"
+	"github.com/ava-labs/avalanchego/utils/wrappers"
+
+	"github.com/ava-labs/hypersdk/codec"
+	"github.com/ava-labs/hypersdk/state"
+	"github.com/ava-labs/hypersdk/x/fdsmr"
+)
+
+const bondAllocSize = 128
+
+var (
+	// ErrMissingBond is fatal and is returned if Unbond is called on a
+	// transaction that was not previously bonded
+	ErrMissingBond = errors.New("missing bond")
+
+	_ fdsmr.Bonder[*Transaction] = (*Bonder)(nil)
+)
+
+type BondBalance struct {
+	Pending uint32 `serialize:"true"`
+	Max     uint32 `serialize:"true"`
+}
+
+// Bonder maintains state of account bond balances to limit the amount of
+// pending transactions per account
+type Bonder struct{}
+
+// this needs to be thread-safe if it's called from the api
```

Comment:
Should this comment be removed since it's writing to `state.Mutable` which is a merkledb view? We never write concurrently to the merkledb views elsewhere